tutor/src/components/notes.js

```diff
@@ -34,6 +34,11 @@ class NotesWidgetWrapper extends React.Component {
     page: PropTypes.instanceOf(Page).isRequired,
   };
 
+  constructor(props) {
+    super(props);
+    props.course.notes.ensurePageExists(props.page);
+  }
+
```

Comment:
I don't think we can call `props.course.notes.ensurePageExists(props.page);` here because it won't run if the props.page updates.  we need to call it before the highlighter is setup in the `initializePage` method, probably right after the `isMounted` check.

Reply:
I think you moved this there because https://github.com/openstax/tutor-js/pull/2841/files#diff-16bdd0ebc9ce6c81cf33368a50fc4fbaL45 requires it, and it uses the api pending check to `initializePage` after mount. 

I had trouble seeing it in testing, but the potential issue you pointed out makes sense to me... maybe we change `NotesWidget` to handle it internally? What are your thoughts about how to best change that?